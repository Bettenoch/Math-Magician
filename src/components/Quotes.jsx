import React, { useEffect, useState } from 'react';
import '../styles/Quotes.scss';

const Quotes = () => {
  const UniqueId = 'FQ9G9AMY8QX/B/b0zKMv7w==oP5aqUqH7SQufVDb';
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(null);
  const [isLoading, setIsLoading] = useState(false);

  const fetchQuote = async () => {
    setIsLoading(true);
    try {
      const res = await fetch('https://api.api-ninjas.com/v1/quotes?category=inspirational', {
        headers: {
          'X-Api-Key': UniqueId,
        },
      });
      const val = await res.json();
      setData(val);
    } catch (error) {
      setHasError('Something went wrong, please refresh the page');
    }
    setIsLoading(false);
  };

  useEffect(() => {
    fetchQuote();
  }, []);

  if (hasError) {
    return (
      <div className="error-popup">
        <p>{ hasError }</p>
      </div>
    );
  }

  if (isLoading) {
    return (
      <article className="loader-container">
        <div className="loader">
          <div className="bar1" />
          <div className="bar2" />
          <div className="bar3" />
          <div className="bar4" />
          <div className="bar5" />
          <div className="bar6" />
        </div>
      </article>
    );
  }
  const refreshQuote = () => {
    fetchQuote();
  };

  return (
    <article data-testid="quote-space" className="quote-container">
      {
            data.length > 0 ? (
              <article className="quote-display">
                <p className="showQuote" data-testid="test-quote">
                  {data[0]?.quote}
                </p>
                <span className="author">
                  {data[0]?.author}
                </span>
              </article>
            ) : (
              <article>
                <p className="showQuote">Please refresh to see some quotes.</p>
              </article>
            )
        }
      <button data-testid="test-refresh" className="quote-btn" type="button" onClick={refreshQuote}>  Refresh</button>
    </article>
  );
};

export default Quotes;
