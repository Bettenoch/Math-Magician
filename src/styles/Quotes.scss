.quote-container {
  display: flex;
  flex-direction: column;
  gap: 2rem;
  max-width: 40%;
  margin-top: 10rem;
}

.quote-display {
  display: flex;
  flex-direction: column;
  padding: 4rem 2rem;
  align-items: center;
  background-image: url('../Assets//images/floral1.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  border-radius: 10px;
  box-shadow: 0 26px 58px 0 rgba(0, 0, 0, 0.22), 0 5px 14px 0 rgba(0, 0, 0, 0.18);

  .showQuote {
    font-size: 1.2rem;
    font-weight: 500;
    text-align: center;
    padding-bottom: 1rem;
  }

  .author {
    color: rgb(41, 76, 96);
    text-decoration: underline;
  }
}

.loader-container {
  position: relative;
  display: flex;
  width: 40%;
  height: 15rem;
  margin-top: 4rem;
  background-image: url('../Assets//images/floral1.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  border-radius: 10px;
  box-shadow: 0 -25px 20px -20px rgba(0, 0, 0, 0.45), 0 25px 20px -20px rgba(0, 0, 0, 0.45), 25px 0 20px -20px rgba(0, 0, 0, 0.45);
}

.loader {
  margin: 0 auto;
  width: 60px;
  height: 50px;
  text-align: center;
  font-size: 10px;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);

  > div {
    height: 100%;
    width: 8px;
    display: inline-block;
    margin-left: 2px;
    -webkit-animation: delay 0.8s infinite ease-in-out;
    animation: delay 0.8s infinite ease-in-out;
  }

  .bar1 {
    background-color: #754fa0;
  }

  .bar2 {
    background-color: #ae0f0c;
    -webkit-animation-delay: -0.7s;
    animation-delay: -0.7s;
  }

  .bar3 {
    background-color: #90d36b;
    -webkit-animation-delay: -0.6s;
    animation-delay: -0.6s;
  }

  .bar4 {
    background-color: #f2d40d;
    -webkit-animation-delay: -0.5s;
    animation-delay: -0.5s;
  }

  .bar5 {
    background-color: #fcb12b;
    -webkit-animation-delay: -0.4s;
    animation-delay: -0.4s;
  }

  .bar6 {
    background-color: #ed1b72;
    -webkit-animation-delay: -0.3s;
    animation-delay: -0.3s;
  }
}

@-webkit-keyframes delay {
  0%,
  40%,
  100% { -webkit-transform: scaleY(0.05); }
  20% { -webkit-transform: scaleY(1); }
}

@keyframes delay {
  0%,
  40%,
  100% {
    transform: scaleY(0.05);
    -webkit-transform: scaleY(0.05);
  }

  20% {
    transform: scaleY(1);
    -webkit-transform: scaleY(1);
  }
}

.quote-btn {
  width: max-content;
  background-color: #03678c8f;
  border-radius: 10px;
  color: aliceblue;
}
