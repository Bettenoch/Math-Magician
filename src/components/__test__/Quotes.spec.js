import renderer from 'react-test-renderer';
import Quotes from '../Quotes';

describe('should display quotes', () => {
  it('should show one comment', () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
