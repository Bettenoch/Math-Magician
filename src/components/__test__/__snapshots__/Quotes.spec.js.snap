// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display quotes should show one comment 1`] = `
<article
  className="quote-container"
  data-testid="quote-space"
>
  <article>
    <p
      className="showQuote"
    >
      Please refresh to see some quotes.
    </p>
  </article>
  <button
    className="quote-btn"
    data-testid="test-refresh"
    onClick={[Function]}
    type="button"
  >
      Refresh
  </button>
</article>
`;
